src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterServices.cs

```diff
@@ -16,7 +16,11 @@ public static class FormatterServices
     {
         private static readonly ConcurrentDictionary<MemberHolder, MemberInfo[]> s_memberInfoTable = new ConcurrentDictionary<MemberHolder, MemberInfo[]>();
 
-        private static FieldInfo[] InternalGetSerializableMembers(Type type)
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:UnrecognizedReflectionPattern",
+            Justification = "The Type is annotated with All, which will preserve base type fields.")]
```

Comment:
Do we want to cut a bug in illink for this? I assume the problem is that accessing `Type.BaseType` returns something without any annotations but if we know the `this` of the `Type` is `.All`, the `.BaseType` should be considered having `.All` as well.

Reply:
We can infer other things about base types as well - for example PublicMethods would propagate as well (actually any Public* would propagate).

Reply:
I'm not sure illink can figure out what this code is doing.

https://github.com/dotnet/runtime/blob/1f736c151de09da58cf1b5f147e6d750778bad69/src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterServices.cs#L39-L55

calls

https://github.com/dotnet/runtime/blob/1f736c151de09da58cf1b5f147e6d750778bad69/src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterServices.cs#L120-L157

This code is putting all the base types into an array and then this method walks over the array. I don't think illink could figure this out.
I didn't want to touch this code and re-write it to use a loop walking up the BaseType all in one method.